armi/physics/neutronics/crossSectionSettings.py

```diff
@@ -72,44 +123,246 @@
+    """
+    Cross section modeling options for a particular XS ID.
+
+    Attributes
+    ----------
+    xsID : str
+        Cross section ID that is two characters maximum (i.e., AA).
+
+    geometry: str
+        The geometry modeling approximation for regions of the core with
+        this assigned xsID. This is required if the ``fileLocation``
+        attribute is not provided. This cannot be set if the ``fileLocation``
+        is provided.
+
+    fileLocation: list of str
+        This should be a list of paths where the cross sections for this
+        xsID can be copied from. This is required if the ``geometry``
+        attribute is provided. This cannot be set if the ``geometry``
+        is provided.
+
+    validBlockTypes: str or None
+        This is a configuration option for how the cross section group manager
+        determines which blocks/regions to manage as part of the same collection
+        for the current xsID. If this is set to ``None`` then all blocks/regions
+        with the current xsID will be considered.
+
+    blockRepresentation : str
+        This is a configuration option for how the cross section group manager
+        will select how to create a representative block based on the collection
+        within the same xsID. See: ``crossSectionGroupManager.BLOCK_COLLECTIONS``.
+
+    driverID : str
+        This is a lattice physics configuration option used to determine which
+        representative block can be used as a "fixed source" driver for another
+        composition. This is particularly useful for non-fuel or highly subcritical
+        regions.
+
+    criticalBuckling : bool
+        This is a lattice physics configuration option used to enable or disable
+        the critical buckling search option.
+
+    nuclideReactionDriver : str
+        This is a lattice physics configuration option that is similar to the
+        ``driverID``, but rather than applying the source from a specific
+        representative block, the neutron source is taken from a single
+        nuclides fission spectrum (i.e., U235). This is particularly useful
+        for configuring SERPENT 2 lattice physics calculations.
+
+    externalDriver : bool
+        This is a lattice physics configuration option that can be used
+        to determine if the fixed source problem is internally driven
+        or externally driven by the ``driverID`` region. Externally
+        driven means that the region will be placed on the outside of the
+        current xsID block/region.
+
+    useHomogenizedBlockComposition : bool
+        This is a lattice physics configuration option that is useful for
+        modeling spatially dependent problems (i.e., 1D/2D). If this is
+        True then the representative block for the current xsID will be
+        be homogenzied. If this is False then the block will be represented
+        in the geometry type selected. This is mainly used for 1D cylindrical
+        problems.
```

Comment:
wait but if it's 1D and homogenized at the same time... I'm confused

Reply:
Its the difference between having multiple rings of control, clad, coolant, control, clad, coolant, ... for the representative block and being driven by external "fuel" and just being a homogeneous mixture of "control/clad/coolant" as a region and then "fuel" as the separate region. This allows for an external spectrum source onto the control region, but without accounting for the spatial self-shielding effects.